board/app/controllers/user_controller.php

```diff
@@ -78,15 +75,132 @@ public function login()
         $this->render($page); 
     }
         
-    public function logout()
-    {
+    public function logout() {
         session_destroy();
         redirect(url('user/login'));
     }
 
-    public function home()
-    {
+    /*
+    * Display user's name, username, and email
+    
+    public function profile() {
+
+        $user = User::getData();
+        $this->set(get_defined_vars());
+    }*/
+
+     /*
+* Display user's name, username, and email
+*/
+public function profile() {
+$user = User::getData($_SESSION['user_id']);
+$this->set(get_defined_vars());
+}
+
+
+    public function home(){
+
+    }
+
+    public function edit(){
+        $params = array(
+            'username' => Param::get('username'),
+            'first_name' => Param::get('first_name'),
+            'last_name' => Param::get('last_name'),
+            'email' => Param::get('email') 
+            );
+
+        $user = new User($params);
+        $page = Param::get('page_next', 'edit');
+
+        switch ($page) {
+            case 'edit':
+            break;
+
+            case 'edit_end':
+                try {
+                     $user->update();
+                     }catch (ValidationException $e) {
+                        $page = 'edit';
+                     }
+                     break;
+                default:
+                    throw new NotFoundException("{$page} is not found");
+                    break;
+                }
+
         $this->set(get_defined_vars());
+        $this->render($page); 
+
     }
+
+    //View all users - user/users.php
+    public function users() {
+       $user_id = Param::get('user_id');
+       $user = User::get($user_id);
+       $users = User::getAllUsers();
+       $this->set(get_defined_vars()); 
+
+    }
+
+    public function others_success(){
+
+    }
+
+    public function others() {
+      
+        $user_id = Param::get('user_id');   
+        $user = new User;
+        $row = User::get($user_id);
```

Comment:
as the original implementation of `User::get()` is incorrect, you may just do this instead of `foreach`
```php
$row = User::get($user_id);
$user = new User($row);
```